src/App.tsx

```diff
@@ -3,10 +3,86 @@ import './App.scss';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
-// import * as goodsAPI from './api/goods';
+import * as goodsAPI from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goods: Good[];
+  hasError: boolean;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+    hasError: false,
+  };
+
+  loadGoods = async (getGoods: () => Promise<Good[]>) => {
+    try {
+      const goods = await getGoods();
+
+      this.setState({ goods });
+    } catch (error) {
+      this.setState({ hasError: true });
+    }
+  };
+
+  handleLoadAllGoods = () => {
+    this.loadGoods(goodsAPI.getAll);
+  };
+
+  handleLoadRedGoods = () => {
+    this.loadGoods(goodsAPI.getRedGoods);
+  };
+
+  handleLoadFirstFiveGoods = () => {
+    this.loadGoods(goodsAPI.get5First);
+  };
```

Comment:
you can pass needed functions directly on onClick and not create all these methods
```onClick={() => this.loadGoods(goodsApi.getAll)}```